OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs

```diff
@@ -150,5 +150,73 @@ protected static bool ReloadsAutomatically(IEnumerable<AmmoPool> ammoPools, Rear
 
 			return true;
 		}
+
+		// Retreat units from combat, or for supply only in idle
+		protected void Retreat(Squad squad, bool flee, bool rearm, bool repair)
+		{
+			var loc = new CPos(0, 0, 0);
+
+			// HACK: "alreadyRepair" is to solve AI repair orders performance,
+			// which is only allow one goes to repairpad at the same time to avoid queueing too many orders.
+			// if repairpad logic is better we can just drop it.
+			var alreadyRepair = false;
+
+			if (flee)
+				loc = RandomBuildingLocation(squad);
```

Comment:
```suggestion
				var loc = RandomBuildingLocation(squad);
```

as it is only used when `flee` is true anyway.

Reply:
Don't get this point, add the `var` here is invalid and I in fact want to save some performance because `RandomBuildingLocation` will have a scan over all units which may be slow and costy.

Reply:
See https://github.com/OpenRA/OpenRA/pull/18039/files/b8434b7443dbaf9caf6829d389e6f9ad555d0a9b#r519184241